phoenix-core/src/main/java/org/apache/phoenix/compile/WhereOptimizer.java

```diff
@@ -155,6 +155,7 @@ public static Expression pushKeyExpressionsToScan(StatementContext context, Set<
             keySlots = whereClause.accept(visitor);
 
             if (keySlots == null && (tenantId == null || !table.isMultiTenant()) && table.getViewIndexId() == null && !minOffset.isPresent()) {
+                // FIXME this overwrites salting info in the scanRange
```

Comment:
which query will be impacted due to the override?

Reply:
None.
If this were fixed, we could take use the salting info from the scanRange object.
Now, we have to go back to the PTable to get it, and the salting info in ScanRange is not used.

Reply:
Currently none.
We just need to re-read some salting info from the table, instead of taking the cached value from

Reply:
got it, sounds good